src/rate/coingecko.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+
+import { type EdgeCorePluginOptions, type EdgeRatePlugin } from 'edge-core-js/types'
+
+export function makeCoinGeckoPlugin(opts: EdgeCorePluginOptions): EdgeRatePlugin {
+  const { io } = opts
+
+  return {
+    rateInfo: {
+      displayName: 'Coincap',
+    },
+
+    async fetchRates(pairsHint) {
+      const reply = await io.fetch(
+        'https://api.coingecko.com/api/v3/coins/telos?localization=false&tickers=false&market_data=true&community_data=false&developer_data=false&sparkline=false'
+      )
+      const json = await reply.json()
+
+      // Grab all the pairs (TLOS to _______)
+      const pairs = []
+      if (json && json.market_data && json.market_data.current_price && json.market_data.current_price.usd) {
```

Comment:
Are you familiar with cleaners? We generally prefer those for JSON validation going forward.

```
import { asObject, asNumber } from 'cleaners'

type asGeckoUsdReply = asObject({
  market_data: asObject({
    current_price: asObject({
      usd: asNumber
    })
  })
})
```

This is like a TypeScript type definition, but it happens at runtime. Now we can use this function to validate the JSON:

```
try {
  const rate = asGeckoUsdReply(json).market_data.current_price.usd
  pairs.push({ ..., rate })
} catch () {}
```